lms/static/scripts/frontend_apps/components/dashboard/AssignmentActivity.tsx

```diff
@@ -78,6 +89,40 @@ export default function AssignmentActivity() {
           {assignment.data && title}
         </h2>
       </div>
+      {assignment.data && (
+        <DashboardActivityFilters
+          courses={{
+            activeItem: assignment.data.course,
+            // When the active course is cleared, navigate to home, but keeping
+            // active assignment and students
+            onClear: () =>
+              navigate(
+                recordToQueryString({
+                  student_id: studentIds,
+                  assignment_id: assignmentId,
+                }),
+              ),
+          }}
+          assignments={{
+            activeItem: assignment.data,
+            // When active assignment is cleared, navigate to its course page,
+            // but keeping other query params intact
+            onClear: () => {
+              const query = search.length === 0 ? '' : `?${search}`;
+              navigate(`${courseURL(assignment.data!.course.id)}${query}`);
```

Comment:
I'm using a non-null assertion operator here, because if `assignment.data`is not set, we won't be rendering filters at all, so it will be set when this is called.

Reply:
This makes sense. TS can't prove `assignment.data` is non-null at the point when `onClear` is invoked because it doesn't have a deep enough understanding of Preact semantics.